example/HMI/views/main.handlebars

```diff
@@ -1 +1,16 @@
+
+<div data-map="{'innerHTML':{'data-var-name':'testData'}}">
+    <h1>Example</h1>
+    <p>Example HMI</p>
+</div>
+
+<input class="lux-text-value" data-var-name="input.color"/>
+<input class="lux-text-value" data-var-name="input.width"/>
+<input id="testInput" 
+data-map="{
+    'value':{'data-var-name':'testData'}, 
+    'style.backgroundColor':'input.color',
```

Comment:
Since the focus of the PR is on working with web components, can you tell me more about the choice to make this custom mapping attribute, vs leveraging slots on web components?

Reply:
I'm not sure how slots are relevant, what do you mean? My understanding is that slots are used when you want to integrate user supplied html into the web component, but this is about setting values on the web component

Here is an example usage with a web component:
```html
<ob-rudder-medium  data-map="{ 'rudderAngle' : 'tmplitTest:tmplit.Slider' , ' rudderSetPointAngle' : 'tmp1itTest:tmplit.Numericlnput' }"/>
```

Reply:
Slots could be relevant for uses like input forms, because (I think) the slot could just be one part of the component, with the binding already applied.
<my-fancy-input>
    <slot name="userInput" data-var-name="PLC:varToWriteTo" />
    <slot name="otherThing" data-var-name="OtherVar"/>
</my-fancy-input>

But on second thought, what is even more relevant than slots is custom attributes on web components. I believe with these you could expose certain component-specific parameters, rather than using an all-in-one mapping. I think it would be much more readable / usable, but would perhaps limit which attributes you could bind to.

Loosely-psuedocoded-example:
`<od-rudder-medium data-rudder="tmplitTest:tmplit.Slider" data-rudder-angle="tmplit1Test.NumericInput"/>`

The biggest downside I see to the mapping, is readability and how easily one could misspell part of the mapping, and then break all of the mapping for that component.

Reply:
The main use case is web-components that we don't build. This is mainly useful for third party components, and means we don't have to consider how the data will map when creating components

Reply:
Note that the od-rudder-medium example is from open-bridge

Reply:
The data-mapping solution offers a lot of control, but also suffers from readability, and requires internal knowledge of the component. The implications for styling seem convenient for some custom overrides, but tricky for others.

I think other frameworks offer much more readable versions of this kind of solution, and it's worth considering how this could be more dev-friendly.

And when we do make our own LUX-specific web components, exposing commonly-bound attribute vars (data-rudder, etc) would be a good practice.

Reply:
I don't see why it requires internal knowledge of the components, the normal use case would likely not use deep structures. You just have to know what properties to set, which would be the case with any solution.

I don't think we should build data binding into our components, IMO they should really be standalone and not consider integration into lux. The data-map solution covers all use cases, although I agree the json string is not ideal.

I was considering something like

data-lux-[arbitrary property] = "my.pv.name" but I'm not sure this is actually very good from an implementation standpoint, because there aren't great built-in ways to get data-lux* for all elements although that might not be correct.

Can you give any examples of more dev-friendly apis?

Reply:
I just tried using document.querySelectorAll('[data-lux*]') and it throws an error that it isn't a valid selector

Reply:
The only way I see that we could do it is with something like
```javascript
let elements = Array.from(document.querySelectorAll('*')).filter(el => 
    Array.from(el.attributes).some(attr => attr.name.startsWith('data-lux'))
);
```
but that is really inefficient

Reply:
I tested this and it's ~100x slower